src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java

```diff
@@ -86,9 +96,41 @@ public List<BlogDto> getBlogsByTagName(String tagName) {
     @Override
     public List<SimplifiedBlogResponse> getSimplifiedBlogs() {
         List<Blog> blogs = blogRepository.findAll();
-        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(),blog.getText())).toList();
+        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(), blog.getText())).toList();
+    }
+
+    @Override
+    public UserResponse deleteUserBlog(Long blogId, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new ResourceNotFound("User Not Found"));
+        Blog blog = user.getBlogList().stream()
+                .filter(b -> b.getId().equals(blogId))
+                .findFirst()
+                .orElseThrow(() -> new ResourceNotFound("Blog does not exist with id: " + blogId));
+        user.getBlogList().remove(blog);
+        User savedUser = userRepository.save(user);
+        blogRepository.delete(blog);
+        UserResponse userResponse = new UserResponse();
+        userResponse.setUserId(savedUser.getId());
+        Set<TagDto> tagDtoSet = blog.getTags().stream().map(t -> new TagDto(t.getId(), t.getName())).collect(Collectors.toSet());
+        userResponse.setBlogList(user.getBlogList().stream()
+                .map(b -> new BlogDto(b.getId(), b.getTitle(), b.getText(), tagDtoSet))
+                .collect(Collectors.toList()));
+        return userResponse;
+
     }
+    @Override
+    public List<UserBlogResponse> getUsersBlogs() {
 
+        return userRepository.findAll().stream().map(user -> {
```

Comment:
I believe this piece of code will fetch all blogs for all users ? We want the functionality to fetch blog by username

Reply:
I thought that user can get all the blogs of all the users. I changed it to get blogs of specific user.